bitcoin/tx_parts.c

```diff
@@ -181,29 +181,48 @@ static struct wally_tx_input *fromwire_wally_tx_input(const tal_t *ctx,
 		blinding_nonce = fromwire_tal_arrn(tmpctx,
 						   cursor, max,
 						   fromwire_u32(cursor, max));
+
+		/* libwally doesn't like non-NULL ptrs with zero lengths. */
```

Comment:
I wonder if we should just make fromwire_tal_arrn return NULL for zero length, instead.  We do that in our generated code already...

Reply:
~That'd be a bit strange if you ask me: when I get an array of `u8` for example I expect to be able to `tal_arr_expand` or `tal_resize` it, which doesn't work for `NULL` (because `tal_count` on `NULL` will likely not work).~

Edit: After checking with `tal.c` it appears that `tal_bytelen(NULL)` and `tal_count(NULL)` do the right thing :tm: :+1: